CI errors for Code refactoring

```
Refactor Rename
       Rename function in source without a trailing empty line:
      AssertionError: expected '--- a/without empty line.py\n+++ b/without empty line.py\n@@ -1,8 +1,8 @@\n import os\n \n-def one():\n+def three():\n     return True\n \n def two():\n-    if one():\n-        print("A" + one())\r\n+    if three():\n+        print("A" + three())\r\n' to equal '--- a/without empty line.py\r\n+++ b/without empty line.py\r\n@@ -1,8 +1,8 @@\r\n import os\r\n \r\n-def one():\r\n+def three():\r\n     return True\r\n \r\n def two():\r\n-    if one():\r\n-        print("A" + one())\r\n+    if three():\r\n+        print("A" + three())\r\n'
      + expected - actual
      ---- a/without empty line.py
      -+++ b/without empty line.py
      -@@ -1,8 +1,8 @@
      - import os
      - 
      --def one():
      -+def three():
      -     return True
      - 
      - def two():
      --    if one():
      +--- a/without empty line.py
      ++++ b/without empty line.py
      +@@ -1,8 +1,8 @@
      + import os
      + 
      +-def one():
      ++def three():
      +     return True
      + 
      + def two():
      +-    if one():
       -        print("A" + one())
      -+    if three():
      ++    if three():
       +        print("A" + three())
      
      at Object.<anonymous> (C:\projects\vscode-python-v3vd6\src\test\refactor\rename.test.ts:58:48)
      at Generator.next (<anonymous>)
      at fulfilled (C:\projects\vscode-python-v3vd6\out\test\refactor\rename.test.js:6:58)
      at <anonymous>
  2) Refactor Rename
       Rename function in source with a trailing empty line:
      AssertionError: expected '--- a/with empty line.py\n+++ b/with empty line.py\n@@ -1,8 +1,8 @@\n import os\n \n-def one():\n+def three():\n     return True\n \n def two():\n-    if one():\n-        print("A" + one())\n+    if three():\n+        print("A" + three())\n' to equal '--- a/with empty line.py\r\n+++ b/with empty line.py\r\n@@ -1,8 +1,8 @@\r\n import os\r\n \r\n-def one():\r\n+def three():\r\n     return True\r\n \r\n def two():\r\n-    if one():\r\n-        print("A" + one())\r\n+    if three():\r\n+        print("A" + three())\r\n'
      + expected - actual
      ---- a/with empty line.py
      -+++ b/with empty line.py
      -@@ -1,8 +1,8 @@
      - import os
      - 
      --def one():
      -+def three():
      -     return True
      - 
      - def two():
      --    if one():
      --        print("A" + one())
      -+    if three():
      -+        print("A" + three())
      +--- a/with empty line.py
      ++++ b/with empty line.py
      +@@ -1,8 +1,8 @@
      + import os
      + 
      +-def one():
      ++def three():
      +     return True
      + 
      + def two():
      +-    if one():
      +-        print("A" + one())
      ++    if three():
      ++        print("A" + three())
      
      at Object.<anonymous> (C:\projects\vscode-python-v3vd6\src\test\refactor\rename.test.ts:70:48)
      at Generator.next (<anonymous>)
      at fulfilled (C:\projects\vscode-python-v3vd6\out\test\refactor\rename.test.js:6:58)
      at <anonymous>

```

--- a/news/3 Code Health/1953.md
+++ b/news/3 Code Health/1953.md
@@ -1,0 +1,1 @@
+Fix rename refactoring unit tests.

--- a/src/client/common/extensions.ts
+++ b/src/client/common/extensions.ts
@@ -15,7 +15,7 @@ declare interface String {
      * By default lines are trimmed and empty lines are removed.
      * @param {SplitLinesOptions=} splitOptions - Options used for splitting the string.
      */
-    splitLines(splitOptions?: { trim: boolean, removeEmptyEntries?: boolean }): string[];
+    splitLines(splitOptions?: { trim: boolean; removeEmptyEntries?: boolean }): string[];
     /**
      * Appropriately formats a string so it can be used as an argument for a command in a shell.
      * E.g. if an argument contains a space, then it will be enclosed within double quotes.
@@ -33,10 +33,10 @@ declare interface String {
  * By default lines are trimmed and empty lines are removed.
  * @param {SplitLinesOptions=} splitOptions - Options used for splitting the string.
  */
-String.prototype.splitLines = function (this: string, splitOptions: { trim: boolean, removeEmptyEntries: boolean } = { removeEmptyEntries: true, trim: true }): string[] {
+String.prototype.splitLines = function (this: string, splitOptions: { trim: boolean; removeEmptyEntries: boolean } = { removeEmptyEntries: true, trim: true }): string[] {
     let lines = this.split(/\r?\n/g);
     if (splitOptions && splitOptions.trim) {
-        lines = lines.filter(line => line.trim());
+        lines = lines.map(line => line.trim());
     }
     if (splitOptions && splitOptions.removeEmptyEntries) {
         lines = lines.filter(line => line.length > 0);

--- a/src/test/refactor/rename.test.ts
+++ b/src/test/refactor/rename.test.ts
@@ -26,7 +26,7 @@ suite('Refactor Rename', () => {
     const options: TextEditorOptions = { cursorStyle: TextEditorCursorStyle.Line, insertSpaces: true, lineNumbers: TextEditorLineNumbersStyle.Off, tabSize: 4 };
     let pythonSettings: typeMoq.IMock<IPythonSettings>;
     let serviceContainer: typeMoq.IMock<IServiceContainer>;
-    const EOLUsedByRoped = '\n';
+    const EOLUsedByRope = '\n';
     suiteSetup(initialize);
     setup(async () => {
         pythonSettings = typeMoq.Mock.ofType<IPythonSettings>();
@@ -47,7 +47,7 @@ suite('Refactor Rename', () => {
 
     test('Rename function in source without a trailing empty line', async () => {
         const sourceFile = path.join(EXTENSION_ROOT_DIR, 'src', 'test', 'pythonFiles', 'refactoring', 'source folder', 'without empty line.py');
-        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOLUsedByRoped}+++ b/${path.basename(sourceFile)}${EOLUsedByRoped}@@ -1,8 +1,8 @@${EOLUsedByRoped} import os${EOLUsedByRoped} ${EOLUsedByRoped}-def one():${EOLUsedByRoped}+def three():${EOLUsedByRoped}     return True${EOLUsedByRoped} ${EOLUsedByRoped} def two():${EOLUsedByRoped}-    if one():${EOLUsedByRoped}-        print(\"A\" + one())${EOLUsedByRoped}+    if three():${EOLUsedByRoped}+        print(\"A\" + three())${EOLUsedByRoped}`;
+        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOLUsedByRope}+++ b/${path.basename(sourceFile)}${EOLUsedByRope}@@ -1,8 +1,8 @@${EOLUsedByRope} import os${EOLUsedByRope} ${EOLUsedByRope}-def one():${EOLUsedByRope}+def three():${EOLUsedByRope}     return True${EOLUsedByRope} ${EOLUsedByRope} def two():${EOLUsedByRope}-    if one():${EOLUsedByRope}-        print(\"A\" + one())${EOLUsedByRope}+    if three():${EOLUsedByRope}+        print(\"A\" + three())${EOLUsedByRope}`;
 
         const proxy = new RefactorProxy(EXTENSION_ROOT_DIR, pythonSettings.object, path.dirname(sourceFile), serviceContainer.object);
         const textDocument = await workspace.openTextDocument(sourceFile);
@@ -59,7 +59,7 @@ suite('Refactor Rename', () => {
     });
     test('Rename function in source with a trailing empty line', async () => {
         const sourceFile = path.join(EXTENSION_ROOT_DIR, 'src', 'test', 'pythonFiles', 'refactoring', 'source folder', 'with empty line.py');
-        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOLUsedByRoped}+++ b/${path.basename(sourceFile)}${EOLUsedByRoped}@@ -1,8 +1,8 @@${EOLUsedByRoped} import os${EOLUsedByRoped} ${EOLUsedByRoped}-def one():${EOLUsedByRoped}+def three():${EOLUsedByRoped}     return True${EOLUsedByRoped} ${EOLUsedByRoped} def two():${EOLUsedByRoped}-    if one():${EOLUsedByRoped}-        print(\"A\" + one())${EOLUsedByRoped}+    if three():${EOLUsedByRoped}+        print(\"A\" + three())${EOLUsedByRoped}`;
+        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOLUsedByRope}+++ b/${path.basename(sourceFile)}${EOLUsedByRope}@@ -1,8 +1,8 @@${EOLUsedByRope} import os${EOLUsedByRope} ${EOLUsedByRope}-def one():${EOLUsedByRope}+def three():${EOLUsedByRope}     return True${EOLUsedByRope} ${EOLUsedByRope} def two():${EOLUsedByRope}-    if one():${EOLUsedByRope}-        print(\"A\" + one())${EOLUsedByRope}+    if three():${EOLUsedByRope}+        print(\"A\" + three())${EOLUsedByRope}`;
 
         const proxy = new RefactorProxy(EXTENSION_ROOT_DIR, pythonSettings.object, path.dirname(sourceFile), serviceContainer.object);
         const textDocument = await workspace.openTextDocument(sourceFile);

--- a/src/test/refactor/rename.test.ts
+++ b/src/test/refactor/rename.test.ts
@@ -4,7 +4,6 @@
 'use strict';
 
 import { expect } from 'chai';
-import { EOL } from 'os';
 import * as path from 'path';
 import * as typeMoq from 'typemoq';
 import { Range, TextEditorCursorStyle, TextEditorLineNumbersStyle, TextEditorOptions, window, workspace } from 'vscode';
@@ -27,6 +26,7 @@ suite('Refactor Rename', () => {
     const options: TextEditorOptions = { cursorStyle: TextEditorCursorStyle.Line, insertSpaces: true, lineNumbers: TextEditorLineNumbersStyle.Off, tabSize: 4 };
     let pythonSettings: typeMoq.IMock<IPythonSettings>;
     let serviceContainer: typeMoq.IMock<IServiceContainer>;
+    const EOLUsedByRoped = '\n';
     suiteSetup(initialize);
     setup(async () => {
         pythonSettings = typeMoq.Mock.ofType<IPythonSettings>();
@@ -47,7 +47,7 @@ suite('Refactor Rename', () => {
 
     test('Rename function in source without a trailing empty line', async () => {
         const sourceFile = path.join(EXTENSION_ROOT_DIR, 'src', 'test', 'pythonFiles', 'refactoring', 'source folder', 'without empty line.py');
-        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOL}+++ b/${path.basename(sourceFile)}${EOL}@@ -1,8 +1,8 @@${EOL} import os${EOL} ${EOL}-def one():${EOL}+def three():${EOL}     return True${EOL} ${EOL} def two():${EOL}-    if one():${EOL}-        print(\"A\" + one())${EOL}+    if three():${EOL}+        print(\"A\" + three())${EOL}`;
+        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOLUsedByRoped}+++ b/${path.basename(sourceFile)}${EOLUsedByRoped}@@ -1,8 +1,8 @@${EOLUsedByRoped} import os${EOLUsedByRoped} ${EOLUsedByRoped}-def one():${EOLUsedByRoped}+def three():${EOLUsedByRoped}     return True${EOLUsedByRoped} ${EOLUsedByRoped} def two():${EOLUsedByRoped}-    if one():${EOLUsedByRoped}-        print(\"A\" + one())${EOLUsedByRoped}+    if three():${EOLUsedByRoped}+        print(\"A\" + three())${EOLUsedByRoped}`;
 
         const proxy = new RefactorProxy(EXTENSION_ROOT_DIR, pythonSettings.object, path.dirname(sourceFile), serviceContainer.object);
         const textDocument = await workspace.openTextDocument(sourceFile);
@@ -59,7 +59,7 @@ suite('Refactor Rename', () => {
     });
     test('Rename function in source with a trailing empty line', async () => {
         const sourceFile = path.join(EXTENSION_ROOT_DIR, 'src', 'test', 'pythonFiles', 'refactoring', 'source folder', 'with empty line.py');
-        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOL}+++ b/${path.basename(sourceFile)}${EOL}@@ -1,8 +1,8 @@${EOL} import os${EOL} ${EOL}-def one():${EOL}+def three():${EOL}     return True${EOL} ${EOL} def two():${EOL}-    if one():${EOL}-        print(\"A\" + one())${EOL}+    if three():${EOL}+        print(\"A\" + three())${EOL}`;
+        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOLUsedByRoped}+++ b/${path.basename(sourceFile)}${EOLUsedByRoped}@@ -1,8 +1,8 @@${EOLUsedByRoped} import os${EOLUsedByRoped} ${EOLUsedByRoped}-def one():${EOLUsedByRoped}+def three():${EOLUsedByRoped}     return True${EOLUsedByRoped} ${EOLUsedByRoped} def two():${EOLUsedByRoped}-    if one():${EOLUsedByRoped}-        print(\"A\" + one())${EOLUsedByRoped}+    if three():${EOLUsedByRoped}+        print(\"A\" + three())${EOLUsedByRoped}`;
 
         const proxy = new RefactorProxy(EXTENSION_ROOT_DIR, pythonSettings.object, path.dirname(sourceFile), serviceContainer.object);
         const textDocument = await workspace.openTextDocument(sourceFile);

--- a/src/test/refactor/rename.test.ts
+++ b/src/test/refactor/rename.test.ts
@@ -4,10 +4,12 @@
 'use strict';
 
 import { expect } from 'chai';
+import { EOL } from 'os';
 import * as path from 'path';
 import * as typeMoq from 'typemoq';
 import { Range, TextEditorCursorStyle, TextEditorLineNumbersStyle, TextEditorOptions, window, workspace } from 'vscode';
 import { EXTENSION_ROOT_DIR } from '../../client/common/constants';
+import '../../client/common/extensions';
 import { BufferDecoder } from '../../client/common/process/decoder';
 import { ProcessService } from '../../client/common/process/proc';
 import { PythonExecutionFactory } from '../../client/common/process/pythonExecutionFactory';
@@ -26,7 +28,6 @@ suite('Refactor Rename', () => {
     const options: TextEditorOptions = { cursorStyle: TextEditorCursorStyle.Line, insertSpaces: true, lineNumbers: TextEditorLineNumbersStyle.Off, tabSize: 4 };
     let pythonSettings: typeMoq.IMock<IPythonSettings>;
     let serviceContainer: typeMoq.IMock<IServiceContainer>;
-    const EOLUsedByRope = '\n';
     suiteSetup(initialize);
     setup(async () => {
         pythonSettings = typeMoq.Mock.ofType<IPythonSettings>();
@@ -47,7 +48,8 @@ suite('Refactor Rename', () => {
 
     test('Rename function in source without a trailing empty line', async () => {
         const sourceFile = path.join(EXTENSION_ROOT_DIR, 'src', 'test', 'pythonFiles', 'refactoring', 'source folder', 'without empty line.py');
-        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOLUsedByRope}+++ b/${path.basename(sourceFile)}${EOLUsedByRope}@@ -1,8 +1,8 @@${EOLUsedByRope} import os${EOLUsedByRope} ${EOLUsedByRope}-def one():${EOLUsedByRope}+def three():${EOLUsedByRope}     return True${EOLUsedByRope} ${EOLUsedByRope} def two():${EOLUsedByRope}-    if one():${EOLUsedByRope}-        print(\"A\" + one())${EOLUsedByRope}+    if three():${EOLUsedByRope}+        print(\"A\" + three())${EOLUsedByRope}`;
+        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOL}+++ b/${path.basename(sourceFile)}${EOL}@@ -1,8 +1,8 @@${EOL} import os${EOL} ${EOL}-def one():${EOL}+def three():${EOL}     return True${EOL} ${EOL} def two():${EOL}-    if one():${EOL}-        print(\"A\" + one())${EOL}+    if three():${EOL}+        print(\"A\" + three())${EOL}`
+            .splitLines({ removeEmptyEntries: false, trim: false });
 
         const proxy = new RefactorProxy(EXTENSION_ROOT_DIR, pythonSettings.object, path.dirname(sourceFile), serviceContainer.object);
         const textDocument = await workspace.openTextDocument(sourceFile);
@@ -55,11 +57,12 @@ suite('Refactor Rename', () => {
 
         const response = await proxy.rename<RenameResponse>(textDocument, 'three', sourceFile, new Range(7, 20, 7, 23), options);
         expect(response.results).to.be.lengthOf(1);
-        expect(response.results[0].diff).to.be.equal(expectedDiff);
+        expect(response.results[0].diff.splitLines({ removeEmptyEntries: false, trim: false })).to.be.deep.equal(expectedDiff);
     });
     test('Rename function in source with a trailing empty line', async () => {
         const sourceFile = path.join(EXTENSION_ROOT_DIR, 'src', 'test', 'pythonFiles', 'refactoring', 'source folder', 'with empty line.py');
-        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOLUsedByRope}+++ b/${path.basename(sourceFile)}${EOLUsedByRope}@@ -1,8 +1,8 @@${EOLUsedByRope} import os${EOLUsedByRope} ${EOLUsedByRope}-def one():${EOLUsedByRope}+def three():${EOLUsedByRope}     return True${EOLUsedByRope} ${EOLUsedByRope} def two():${EOLUsedByRope}-    if one():${EOLUsedByRope}-        print(\"A\" + one())${EOLUsedByRope}+    if three():${EOLUsedByRope}+        print(\"A\" + three())${EOLUsedByRope}`;
+        const expectedDiff = `--- a/${path.basename(sourceFile)}${EOL}+++ b/${path.basename(sourceFile)}${EOL}@@ -1,8 +1,8 @@${EOL} import os${EOL} ${EOL}-def one():${EOL}+def three():${EOL}     return True${EOL} ${EOL} def two():${EOL}-    if one():${EOL}-        print(\"A\" + one())${EOL}+    if three():${EOL}+        print(\"A\" + three())${EOL}`
+            .splitLines({ removeEmptyEntries: false, trim: false });
 
         const proxy = new RefactorProxy(EXTENSION_ROOT_DIR, pythonSettings.object, path.dirname(sourceFile), serviceContainer.object);
         const textDocument = await workspace.openTextDocument(sourceFile);
@@ -67,6 +70,6 @@ suite('Refactor Rename', () => {
 
         const response = await proxy.rename<RenameResponse>(textDocument, 'three', sourceFile, new Range(7, 20, 7, 23), options);
         expect(response.results).to.be.lengthOf(1);
-        expect(response.results[0].diff).to.be.equal(expectedDiff);
+        expect(response.results[0].diff.splitLines({ removeEmptyEntries: false, trim: false })).to.be.deep.equal(expectedDiff);
     });
 });

